src/stores/right-panel/RightPanelStoreIPanelState.ts

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { MatrixEvent } from "matrix-js-sdk/src/models/event";
+import { User } from "matrix-js-sdk/src/models/user";
+import { Room } from "matrix-js-sdk/src/models/room";
+import { RoomMember } from "matrix-js-sdk/src/models/room-member";
+import { VerificationRequest } from "matrix-js-sdk/src/crypto/verification/request/VerificationRequest";
+
+import { GroupMember } from "../../components/views/right_panel/UserInfo";
+import { RightPanelPhases } from "./RightPanelStorePhases";
+
+export interface IRightPanelCardState {
+    member?: RoomMember | User | GroupMember;
+    verificationRequest?: VerificationRequest;
+    verificationRequestPromise?: Promise<VerificationRequest>;
+    // group
+    groupId?: string;
+    groupRoomId?: string;
+    widgetId?: string;
+    spaceId?: string;
+    // Room3pidMemberInfo, Space3pidMemberInfo,
+    memberInfoEvent?: MatrixEvent;
+    // threads
+    threadHeadEvent?: MatrixEvent;
+    initialEvent?: MatrixEvent;
+    isInitialEventHighlighted?: boolean;
+}
+
+export interface IRightPanelCardStateStored {
+    member?: RoomMember | User | GroupMember;
+    verificationRequest?: VerificationRequest;
+    verificationRequestPromise?: Promise<VerificationRequest>;
```

Comment:
As you mentioned, I don't think it's actually safe to store these complex objects directly... They won't survive the round trip.

For members, we can save perhaps which type it is and the MXID. For verification requests, we'll need to look at the minimum data needed to rebuild it, or else perhaps leave them out for now.

Reply:
I think leaving them out might be a good solution. The verification probably does not continue working if you restart element anyways?

Reply:
I believe the crypto code will rebuild them from timeline events even across reloads, and all that should be needed to recover them is room ID, as shown in this snippet:

https://github.com/matrix-org/matrix-react-sdk/blob/f4a94f6e7f18c9fd5a2fd65bae56c60ee413299e/src/verification.ts#L126-L132

So perhaps the verification UI can call a code path similar to this, and then there's no need to store them in right panel state?

Reply:
this would be really nice. I think the goal should be to keep the right panel state as minimal as possible.

Reply:
But i think we cannot do this with the verification promise?

Reply:
For the promise, I think we would just drop that and not try to store it, as it's representing in-memory progress anyway.

Reply:
This is now implemented, the store does not contain any verification information anymore and reloads the verification session and the roomMember information on demand.